### ISSUE TITLE
Additional callbacks

### ISSUE THREAD

**Issue by pallas** [nodejs/llhttp#74]
It would be nice to have callbacks after major/minor are set, after header field name is complete, and after header value is complete even if there are more headers. Right now, it's easy enough to emulate, e.g. building up a header field and then on the first header value call recognizing that the field name is complete. However, I'm interested in..

1. building up a URL, possibly 1 character at a time,
2. once the URL is complete, selecting a predefined list of headers I'm interested in,
3. figuring out if the current header name, possibly 1 character at a time, is in this set, &
4. if so, building up it's value, also 1 character at a time; or if not, ignoring the value.

Adding these callbacks would simplify things, e.g. by pulling header field name logic out of the header value callback, &c. This Issue is a pin, since I'm thinking about how to implement this myself. However, I wanted to get feedback on the idea before I started.
